src/coreclr/vm/codeman.cpp

```diff
@@ -3532,7 +3661,52 @@ void EEJitManager::RemoveCodeHeapFromDomainList(CodeHeap *pHeap, LoaderAllocator
     }
 }
 
-void EEJitManager::FreeCodeMemory(HostCodeHeap *pCodeHeap, void * codeStart)
+#ifdef FEATURE_INTERPRETER
+
+InterpreterJitManager::InterpreterJitManager() :
+    m_interpreter(NULL),
+    m_interpreterHandle(NULL),
+    m_interpreterLoadCritSec(CrstSingleUseLock)
+{
+    LIMITED_METHOD_CONTRACT;
+    SetCodeManager(ExecutionManager::GetInterpreterCodeManager());
+}
+
+BOOL InterpreterJitManager::LoadInterpreter()
+{
+    STANDARD_VM_CONTRACT;
+
+    // If the JIT is already loaded, don't take the lock.
+    if (IsInterpreterLoaded())
+        return TRUE;
+
+    // Use m_interpreterLoadCritSec to ensure that the interpreter is loaded on one thread only
+    CrstHolder chRead(&m_interpreterLoadCritSec);
+
+    // Did someone load the JIT before we got the lock?
+    if (IsInterpreterLoaded())
+        return TRUE;
+
+    m_storeRichDebugInfo = CLRConfig::GetConfigValue(CLRConfig::UNSUPPORTED_RichDebugInfo) != 0;
+
+    ICorJitCompiler* newInterpreter = NULL;
+    m_interpreter = NULL;
+
+    g_interpreterLoadData.jld_id = JIT_LOAD_MAIN;
```

Comment:
Why is it `JIT_LOAD_MAIN` instead of a new `JIT_LOAD_INTERP`? I see `JIT_LOAD_ALTJIT` exists. (I don't know the significance of jld_id, I can't find any code that actually reads it, so maybe it doesn't matter?)

Reply:
I think this is just a debugging mean, I'll need to ask JIT folks to get a definitive answer. Anyways, I will add a new JIT_LOAD_variant.